flow/src/main/java/flow/History.java

```diff
@@ -207,7 +207,12 @@ private Builder(Collection<Entry> history) {
     }
 
     public Builder clear() {
-      history.clear();
+      // Clear by popping everything (rather than just calling history.clear()) to
+      // fill up entryMemory. Otherwise we drop view state on the floor.
+      while (!isEmpty()) {
```

Comment:
That is wildly out of scope for this PR. pop() has worked this way for months, so that the builder preserves viewstate. I'm just making the clear() semantics consistent with that. 

See 361541d864207c6e858aab6e00dc6ab61c59ae20 for context.